### DISCUSSION TITLE
GetContentAsAsync - Add CancellationToken

### DISCUSSION THREAD

**Discussion by bednart** [reactiveui/refit#1587] [Ideas]
Hi, is there any reason why there is no CancellationToken argument in [GetContentAsAsync](https://github.com/reactiveui/refit/blob/e726d19e85676097362bd4f3c0b5af2f77512b42/Refit/ApiException.cs#L87) method?

FromHttpContentAsync method which is called inside accepts CancellationToken.

In my opinion GetContentAsAsync should look like this.

```csharp
public async Task<T?> GetContentAsAsync<T>(CancellationToken cancellationToken = default) => HasContent ?
  await RefitSettings.ContentSerializer.FromHttpContentAsync<T>(new StringContent(Content!), cancellationToken).ConfigureAwait(false) :
  default;
```

Do you agree?

Thanks,
beda

**Comment by anaisbetts** [comment]
I'm like 85% sure that the content of the HTTP request has already been downloaded here afaik, there is no practical reason to support cancellation because it's effectively just parsing a buffer. Error responses are also typically very small, and if they're not, I would argue that the server is Doing It Wrong

**Reply by bednart** [reply]
Yeah error responses should be small but still I think it would be better to add `CancellationToken cancellationToken = default` as argument add pass it to FromHttpContentAsync method. Even for that remaining 15 % :) And if you dont want to use it you do not have to.

**Reply by anaisbetts** [reply]
I disagree!
